_docs/tasks/consul.md

```diff
@@ -0,0 +1,101 @@
+---
+title: Running Istio in Docker with Consul
+overview: Make Istio run in docker environment by integrating Consul as a service registry.
+
+order: 42
+
+layout: docs
+type: markdown
+---
+
+Configure Istio to run in docker environment by integrating Consul as a service registry.
+
+## Before you begin
+
+* Setup Istio by following the instructions in the [Installation guide](./installing-istio.html).  You will
+  need `kubectl` and Steps 1-4 from `Installation step` only for this Task.
+
+* Install [Docker](https://docs.docker.com/engine/installation/#cloud) and 
+  [Docker-Compose](https://docs.docker.com/compose/install/)
+
+## Overview
+
+This sample deploys the [BookInfo]({{home}}/docs/samples/bookinfo.html) application. The Consul platform 
+adapter uses Consul Server to help Istio monitor service instances running in the underlying platform. 
+When a service instance is brought up in docker, the [Registrator](http://gliderlabs.github.io/registrator/latest/) 
+automatically registers the service in Consul.
+
+Note that Istio pilot is running inside each app container so as to coordinate Envoy and the service mesh.
+
```

Comment:
should we say Istio pilot agent here?  I think Istio pilot is confusing as not entire pilot runs inside of each app container, correct?